test/integration/request-flow/pm-send-request-cookies.test.js

```diff
@@ -0,0 +1,488 @@
+                });
+
+                it('must have completed the run', function () {
+                    expect(testrun).be.ok();
+                    expect(testrun.done.calledOnce).be.ok();
+                    expect(testrun.done.getCall(0).args[0]).to.be(null);
+                    expect(testrun.start.calledOnce).be.ok();
+                    expect(testrun.io.calledTwice).to.be(true);
+                    expect(testrun.request.calledTwice).to.be(true);
+                });
+
+                it('should handle cookies correctly', function () {
+                    var reqOne = testrun.io.firstCall.args[4],
+                        reqTwo = testrun.request.secondCall.args[3],
+                        resOne = testrun.io.firstCall.args[3];
+
+                    expect(reqOne.headers.reference).to.not.have.property('cookie');
+                    expect(reqTwo.headers.reference.cookie.value).to.contain('foo=bar');
+
+                    expect(resOne.headers.reference['set-cookie'][0]).to.have.property('value', 'foo=bar; Path=/');
+                });
+            });
+        });
+    });
+
+    describe('to sandbox', function () {
+        describe('explicit', function () {
+            describe('clear', function () {
+                var testrun;
+
+                before(function(done) {
+                    this.run({
+                        requester: {
+                            followRedirects: false
+                        },
+                        collection: {
+                            item: {
+                                // ensure that we run something for test and pre-req scripts
+                                event: [{
+                                    listen: 'test',
+                                    script: {
+                                        exec: `
+                                        pm.test('should return cookies correctly', function () {
+                                            pm.response.to.have.jsonBody({ cookies: { foo: 'bar' } });
+                                            pm.expect(pm.cookies.get('foo')).to.be('bar');
+                                        });
+
+                                        pm.cookies.clear();
+
+                                        var sdk = require('postman-collection'),
+                                            req = new sdk.Request('${cookieUrl}');
+
+                                        pm.sendRequest(req, function(err, _response) {
+                                            pm.test('request was sent from sandbox', function () {
+                                                pm.expect(_response).to.have.property('code', 200);
+                                                pm.expect(_response).to.have.property('status', 'OK');
+                                                pm.expect(_response.json).to.equal({ cookies: {} });
+                                                pm.expect(pm.cookies.has('foo')).to.be(false);
```

Comment:
@kunagpal shouldn't we make sure these assertions pass, on `testRun.test.getCall(0)` or `testRun.assertion.getCall(3)`, in order to make sure our tests fail for failures in sandbox scripts?